### ISSUE TITLE
appium server Relay as selenium grid 4 directly to register to grid 4

### ISSUE THREAD

**Issue by krishtoautomate** [appium/appium#17984]
**Is your feature request related to a problem? Please describe.**
Step 1 : appium server need to be started with config/yaml file and 
Step 2 : relay node need to started to connect to selenium-grid-4, 
is it possible to avoid step 2 and make Step 1 run relay as well

**Describe the solution you'd like**
Start Appium-server with yaml file with included configuration of grid ip and port should relay connection to grid similar to grid-4 relay feature

**Comment by jlipps** [comment]
Yes, we would like Appium to work as a grid 4 node in the future, but currently it does not do that. For now, use the strategy described [here](https://appium.github.io/appium/docs/en/latest/guides/grid/)

**Comment by mykola-mokhnach** [comment]
AFAIK Grid 4 does not have any public documentation about how to make that possible. For now the only way to do it would be to go through Grid source code and extract the actual protocol definition from it. Which I find a way complicated task unless one is both Grid and Appium core developer.
@diemol correct me if I'm wrong.

**Comment by diemol** [comment]
That would also create strong dependencies between both servers. So far, the relay approach simplifies the integration.

**Comment by jlipps** [comment]
at the recent selenium+appium summit we did identify this as a convenient way to help the projects integrate. it probably would look the maintenance and documentation of a grid node protocol. it might be worth it though to make things easier to set up for users! interested in your feedback @titusfortner , please cc in anyone else who might have thoughts too

**Comment by titusfortner** [comment]
@diemol and @pujagani are the two who need to be looped into this.

What did we get working at the summit, and what still needs to be done?

**Comment by pujagani** [comment]
I am happy to look into things but this area is new. I will need more context around this to understand it better. @diemol How can I help contribute here?

**Comment by jlipps** [comment]
> What did we get working at the summit

Grid wasn't a target for work at the summit, so nothing

> what still needs to be done

My idea was simply that if the Selenium team could provide the Appium team with some kind of reference for how grid nodes ought to behave, we could add a grid node mode to Appium so that it could connect with the grid hub without needing to run a relay process per Appium server

**Comment by diemol** [comment]
Right, so on the Grid side there not much to do in the code. As @jlipps says, the work that needs to be done is to document how the Nodes send and receive messages through the bus, and more importantly, what messages (including their structure), what endpoints are used by the Distributor and the Router to interact with the Nodes.

All in all, it is possible, but it would tie both servers a bit too much to each other. It is not something we cannot achieve, probably both projects would need to work much closer (at least in this aspect).
